plugins/sniffer.yaml

```diff
@@ -0,0 +1,48 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: sniffer
+spec:
+  version: "v0.0.1"
+  platforms:
+  - selector:
+      matchLabels:
+        os: linux
+        arch: amd64
+    uri: https://github.com/sunny0826/kubectl-sniffer/releases/download/v0.0.1/sniffer_linux_amd64.tar.gz
+    sha256: "13499ebe08ff7880ce2c6a83ae3981086a782e785c71c20a725b99256fa3578a"
+    files:
+    - from: "./sniffer"
+      to: "."
+    - from: LICENSE
+      to: "."
+    bin: "sniffer"
+  - selector:
+      matchLabels:
+        os: darwin
+        arch: amd64
+    uri: https://github.com/sunny0826/kubectl-sniffer/releases/download/v0.0.1/sniffer_darwin_amd64.tar.gz
+    sha256: "f9ec5b523612e82bb944c216b0d55a78a9657cbb42526bada7ea642280f30caf"
+    files:
+    - from: "./sniffer"
+      to: "."
+    - from: LICENSE
+      to: "."
+    bin: "sniffer"
+  - selector:
+      matchLabels:
+        os: windows
+        arch: amd64
+    uri: https://github.com/sunny0826/kubectl-sniffer/releases/download/v0.0.1/sniffer_windows_amd64.zip
+    sha256: "100c6c058cc041b41cbaec728ddc303511366ef5474a0549a98b809c7ee2b8ca"
+    files:
+    - from: "/sniffer.exe"
+      to: "."
+    - from: LICENSE
+      to: "."
+    bin: "sniffer.exe"
+  shortDescription: Sniffing pod-related resources
+  homepage: https://github.com/sunny0826/kubectl-sniffer
+  description: |
+    Find related workloads, namespace, node, service, configmap, secret, 
```

Comment:
Can you elaborate a bit what it means for resources to be "related"? This is not obvious to me, but it seems important to know.

Reply:
Information about the pod's own dependent resources: namespace, node, container, and its mounted configmap and secret, as well as resources that have the same `label` as the pod and serve it: servie, ingress, HPA, etc.

just like:
```
k pod-lens all-logs-fluentd-6
└─┬ [Namespace]  kube-system
  └─┬ [statefulset]  all-logs-fluentd                           Replica: 8/8
    ├─┬ [Node]  ip-172-31-49-23.cn-north-1.compute.internal     [Ready] Node IP: 172.31.49.23
    │ └─┬ [Pod]  all-logs-fluentd-6                            [Running] Pod IP: 100.99.78.33
    │   ├── [Container]  config-reloader                       [Running] Restart: 0
    │   └── [Container]  fluentd                               [Running] Restart: 0
    ├── [PVC]  all-logs-fluentd-buffer-all-logs-fluentd-6
    ├── [Secret]  all-logs-fluentd
    ├── [Secret]  all-logs-fluentd-app
    ├── [Secret]  all-logs-fluentd-output
    └── [Secret]  all-logs-fluentd-token-g9d82

 Related Resources

Kind:         	StatefulSet
Name:         	all-logs-fluentd
Replicas:     	8
---           	---
Kind:         	Service
Name:         	all-logs-fluentd
Cluster IP:   	100.69.216.134
Ports
              	---
              	Name: tcp-fluentd
              	Port: 24240
              	TargetPort: 24240
              	---
              	Name: udp-fluentd
              	Port: 24240
              	TargetPort: 24240
---           	---
Kind:         	PVC
Name:         	all-logs-fluentd-buffer-all-logs-fluentd-0
Storage Class:	gp2
Access Modes: 	ReadWriteOnce
Size:         	20Gi
PV Name:      	pvc-1db4d98a-cec7-4c87-b949-2848a2eb79f7
---           	---
Kind:         	PVC
Name:         	all-logs-fluentd-buffer-all-logs-fluentd-1
Storage Class:	gp2
Access Modes: 	ReadWriteOnce
Size:         	20Gi
PV Name:      	pvc-83ee3807-20e8-4080-bd5d-4bd703835f8c
---           	---
Kind:         	PVC
Name:         	all-logs-fluentd-buffer-all-logs-fluentd-2
Storage Class:	gp2
Access Modes: 	ReadWriteOnce
Size:         	20Gi
PV Name:      	pvc-3404ff1d-bbee-47ab-81b4-31f10368eef2
---           	---
Kind:         	PVC
Name:         	all-logs-fluentd-buffer-all-logs-fluentd-3
Storage Class:	gp2
Access Modes: 	ReadWriteOnce
Size:         	20Gi
PV Name:      	pvc-f9d1ad7e-5d52-4fc2-90ae-bbf44f34ee76
---           	---
Kind:         	PVC
Name:         	all-logs-fluentd-buffer-all-logs-fluentd-4
Storage Class:	gp2
Access Modes: 	ReadWriteOnce
Size:         	20Gi
PV Name:      	pvc-495e41be-73b7-4a5d-b8c9-81e2aeef379a
---           	---
Kind:         	PVC
Name:         	all-logs-fluentd-buffer-all-logs-fluentd-5
Storage Class:	gp2
Access Modes: 	ReadWriteOnce
Size:         	20Gi
PV Name:      	pvc-70a6da49-a9ca-4e9a-8964-c76d8ffa30e5
---           	---
Kind:         	PVC
Name:         	all-logs-fluentd-buffer-all-logs-fluentd-6
Storage Class:	gp2
Access Modes: 	ReadWriteOnce
Size:         	20Gi
PV Name:      	pvc-6f5b7577-f09a-413f-9fab-54e84ad2b638
---           	---
Kind:         	PVC
Name:         	all-logs-fluentd-buffer-all-logs-fluentd-7
Storage Class:	gp2
Access Modes: 	ReadWriteOnce
Size:         	20Gi
PV Name:      	pvc-fe8cbe07-39d7-4593-b075-f70c7099c4c6
---           	---
Kind:         	Secrets
Name:         	all-logs-fluentd
---           	---
Kind:         	Secrets
Name:         	all-logs-fluentd-app
---           	---
Kind:         	Secrets
Name:         	all-logs-fluentd-configcheck-47774f7d
---           	---
Kind:         	Secrets
Name:         	all-logs-fluentd-configcheck-a23682d8
---           	---
Kind:         	Secrets
Name:         	all-logs-fluentd-configcheck-output-47774f7d
---           	---
Kind:         	Secrets
Name:         	all-logs-fluentd-configcheck-output-a23682d8
---           	---
```